IM_calculation/Advanced_IM/overview_adv_IM_status.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import argparse
+import glob
+import pandas as pd
+from pathlib import Path
+import sys
+
+ALL_STATUS = [
+    "finished",
+    "not_started",
+    "not_converged",
+    "not_finished",
+    "timed_out",
+    "crashed",
+    "unknown",
+]
+TYPES = ["obs", "sim"]
+
+SIM_DIR_SUFFIX = "Runs/*/*/IM_calc/{}_status.csv"
+OBS_DIR_SUFFIX = "ObservedGroundMotions/IM_calc/*/{}_status.csv"
+
+
+def process(type, model, path, outdir):
+    event_status_files = glob.glob(str(path))
+
+    if len(event_status_files) == 0:
+        print(f"Error:No status files found at {path}")
+        print(f"Check the root_path and model")
+        sys.exit(0)
+
+    event_status_dict = {}
+    for event_status_file in event_status_files:
+        if type == "sim":
+            event_name = Path(event_status_file).parents[2].name
+        else:
+            event_name = Path(event_status_file).parent.name
+        event_status_df = pd.read_csv(event_status_file)
+        event_status_dict[event_name] = {"total": 0}
+        for st in ALL_STATUS:
+            try:
+                count = event_status_df["status"].value_counts()[st]
+            except KeyError:
+                count = 0
+            event_status_dict[event_name][st] = count
+            event_status_dict[event_name]["total"] += count
+
+    overall_status_df = pd.DataFrame.from_dict(
+        event_status_dict, orient="index", columns=["total"] + ALL_STATUS
+    )
+    overall_status_df.to_csv(outdir / f"overall_status_{model}_{type}.csv")
+    print(overall_status_df)
```

Comment:
Is this print meant to be here?

Reply:
my guess is its intentional. As it would show the user a overall status without the need to open the csv file.

Reply:
It's the whole dataframe? I guess it won't be many lines for smaller runs? But for small mag, it would be like 500 lines?

Reply:
Perhaps something like describe() would be more useful here. With the count for each column total, so if nothing's wrong you don't even need to look in the output file.

Reply:
This "print" is intentional - it also produces a CSV file and it will be more useful if there are lots of lines, but for a large/moderate mag, it saves opening a CSV file - the user requested this feature downloads CSV file and opens in Excel. Sure - Will add some fancy summary

Reply:
@jpa198 
Perhaps this is not what you want to see.
```
>>> df.describe().loc["count"]
total            14.0
finished         14.0
not_started      14.0
not_converged    14.0
not_finished     14.0
timed_out        14.0
crashed          14.0
unknown          14.0
```
My fancy output is this:
```
(python3_maui) baes@maui02: ~$ python overview_adv_IM_status.py /nesi/nobackup/nesi00213/RunFolder/vlo23/first_release_210623_ATC_Steel_SCBF_3Story ATC_Steel_SCBF_3Story sim
-------
1 incomplete/unsuccessful event(s) found (Total: 14)
             total  finished  not_started  not_converged  not_finished  timed_out  crashed  unknown
2016p858000    302       301            0              1             0          0        0        0
(python3_maui) baes@maui02: ~$ python overview_adv_IM_status.py /nesi/nobackup/nesi00213/RunFolder/vlo23/first_release_210623_ATC_Steel_SCBF_3Story ATC_Steel_SCBF_3Story obs
-------
All completed successfully! (Total: 14)
```
I actually considering removing CSV output altogether. I think this terminal output should be good enough

Reply:
I'm probably missing something from that output above how does a user tell which runs need attention?

Reply:
**2016p858000**

Reply:
Ahhhh, I thought that was the name of the run rather than the event. If there are multiple do they appear in a list?

Reply:
Yes, each incomplete, unsuccessful event per each row

Reply:
So if each event in small mag has an issue, the user will still get bombarded with 500 lines of issues in their terminal?

Reply:
That first image from the three above is what I was pushing for. Just something to say something went wrong vs everything's fine, depending on the status counts. Then the user can read the full file to find the issues without their terminal getting swamped

Reply:
yes, and it means the run has some serious issue. `|head N` or `|tail N` can help

Reply:
That's a good point. Seems like the count of each status type is a good output summary and then per-realisation information in the CSV is the next debug step.

Reply:
First of all, let see what James' preferred summary is about.
```
>>> df.describe().loc["count"]
total            14.0
finished         14.0
not_started      14.0
not_converged    14.0
not_finished     14.0
timed_out        14.0
crashed          14.0
unknown          14.0
```
There are 14 events, hence 14 records for each column, which gives no value to the user. Perhaps you are referring to this?
```
>>> df.describe().loc['max']
total            302.0
finished         301.0
not_started        0.0
not_converged      1.0
not_finished       0.0
timed_out          0.0
crashed            0.0
unknown            0.0
Name: max, dtype: float64
```

BTW, this is the full df
```
>>> df
             total  finished  not_started  not_converged  not_finished  timed_out  crashed  unknown
2103645         14        14            0              0             0          0        0        0
2013p543824     86        86            0              0             0          0        0        0
3468575         48        48            0              0             0          0        0        0
3366146        108       108            0              0             0          0        0        0
2016p858000    302       301            0              1             0          0        0        0
2839343         10        10            0              0             0          0        0        0
3124785         34        34            0              0             0          0        0        0
2014p051675     22        22            0              0             0          0        0        0
3631380         58        58            0              0             0          0        0        0
2013p613797     94        94            0              0             0          0        0        0
2015p012816      8         8            0              0             0          0        0        0
3528839         40        40            0              0             0          0        0        0
2808298         18        18            0              0             0          0        0        0
2016p661332     14        14            0              0             0          0        0        0
```

The user sees that there is 1 non_converged case, but the user doesn't know which event it is. So the user has to open the overview CSV, and find it is 2016p858000. Then goes to the event's own CSV for further debug.

I'm thinking of removing the overview CSV altogether. 
If there are many lines and the screen is swamped, it means the whole thing is screwed or it is too early to have majority of runs in "finished" state.  `head` or `tail` commands can come in handy. 
The debug step makes sense when there are only *a few* unsuccessful/incomplete events, and having the directly relevant info on the screen saves the extra step. 

If you have seen the user (who requested this feature) downloading the CSV file from mobaxterm and opens it with Excel, I will find it easier to convince you that my way of "summary" works better.

Reply:
Our concern is that it works for a small number of events. But if you are running it with small mag ~500 events, that becomes unweildly on a terminal. 

I initially thought total was for 14 events total, rather than 14 stations within the event. 

I think a summary of the entire run is most useful. Most of the time we want an overview summary and then ability to go into things if the need arises.

Reply:
ok - I can compromise. By default, it will just print the summary and store the extra info in csv. With extra switch --screen (default false), it will give the summary I like.

Reply:
I do like the idea of having a verbose optional output.

I think it's standard to call it `-v`, `--verbose` however.

Reply:
Ah, my mistake, I thought the "count" would give the sum of each column for some reason. Having the sum in the output would be most useful as you can then go through the csv ticking them off as you go. 
Having the full summary is fine for a handful of events, but definitely not small magnitude (let alone using perturbations). Also having the file lets you go back and check if you need to, vs the transient nature of terminal display

Reply:
--verbose added. Writes to csv as it used to and prints only
```
------------
total            856
finished         855
not_started        0
not_converged      1
not_finished       0
timed_out          0
crashed            0
unknown            0
------------
1 incomplete/unsuccessful event(s) found (Total: 14)
```
but no event info unless --verbose is specified.